### ISSUE TITLE
Authorized_key does not update SELinux context

### ISSUE THREAD

**Issue by c0rner** [ansible/ansible#1762]
When updating keys using the authorized_key module the context of '.ssh' and '.ssh/authorized_keys' is not updated.

Looking at the module code the new athorized_keys file is moved into '.ssh' thus retaining invalid context type.

This can be fixed using notify and a handler using the 'file' module.  This is however a bit inconvenient then using lists for users.

Commit used: 959a46108000b876e4381f5efe792d9df4f5285d

**Comment by dhozac** [comment]
Are you performing the initial creation of the directory/file with authorized_key? That will indeed not work, but updates seem to be behaving as expected, getting the proper context.

**Comment by c0rner** [comment]
Yes, this is used when deploying fresh new servers.  You would expect the context to be updated regardless of an update or a create.  

The authorized_key module is specialized at handling ssh keys and should make sure to keep a sane mode and context on files.

I did take a look at the code to maybe create a "quick" patch but being unfamiliar with python and the current design pattern I decided to wait until I know of a proper fix.
